src/plans/conjugate_gradient_plan.jl

```diff
@@ -507,3 +508,75 @@ function (u::SteepestDirectionUpdateRule)(
 )
     return 0.0
 end
+
+@doc raw"""
+    ConjugateGradientRestart <: DirectionUpdateRule
+
+An update rule might require a restart, that is one gradient step, if the last two gradients
+are nearly orthogonal, cf. [^HagerZhang2006], page 12 (in the pdf, 46 in Journal page numbers).
```

Comment:
This paper isn't linked anywhere (though there is `HagerZhang2005`).

Reply:
HagerZhang2005 is the HZ-Coeff paper linked in the HZ docs. The 2006 one is given in the same string below and is an overview article.

Reply:
Right, I didn't notice.

Reply:
...I actulally also checked 3 times that they were not the same year and not the same ref. ;)